datafusion/physical-expr/src/equivalence.rs

```diff
@@ -231,26 +266,42 @@ mod tests {
     }
 
     #[test]
-    fn merge_equivalence_properties_with_alias_test() -> Result<()> {
-        let mut eq_properties = EquivalenceProperties::new();
-        let mut alias_map = HashMap::new();
-        alias_map.insert(
-            Column::new("a", 0),
-            vec![Column::new("a1", 1), Column::new("a2", 2)],
-        );
+    fn project_equivalence_properties_test() -> Result<()> {
+        let input_schema = Arc::new(Schema::new(vec![
+            Field::new("a", DataType::Int64, true),
+            Field::new("b", DataType::Int64, true),
+            Field::new("c", DataType::Int64, true),
+        ]));
 
-        eq_properties.merge_properties_with_alias(&alias_map);
-        assert_eq!(eq_properties.classes().len(), 1);
-        assert_eq!(eq_properties.classes()[0].len(), 3);
+        let mut input_properties = EquivalenceProperties::new(input_schema);
+        let new_condition = (&Column::new("a", 0), &Column::new("b", 1));
+        input_properties.add_equal_conditions(new_condition);
+        let new_condition = (&Column::new("b", 1), &Column::new("c", 2));
+        input_properties.add_equal_conditions(new_condition);
+
+        let out_schema = Arc::new(Schema::new(vec![
+            Field::new("a1", DataType::Int64, true),
+            Field::new("a2", DataType::Int64, true),
+            Field::new("a3", DataType::Int64, true),
+            Field::new("a4", DataType::Int64, true),
+        ]));
 
         let mut alias_map = HashMap::new();
         alias_map.insert(
             Column::new("a", 0),
-            vec![Column::new("a3", 1), Column::new("a4", 2)],
+            vec![
+                Column::new("a1", 0),
+                Column::new("a2", 1),
+                Column::new("a3", 2),
+                Column::new("a4", 3),
+            ],
         );
-        eq_properties.merge_properties_with_alias(&alias_map);
-        assert_eq!(eq_properties.classes().len(), 1);
-        assert_eq!(eq_properties.classes()[0].len(), 5);
+        let mut out_properties = EquivalenceProperties::new(out_schema);
+
+        project_equivalence_properties(input_properties, &alias_map, &mut out_properties);
+        assert_eq!(out_properties.classes().len(), 1);
+        assert_eq!(out_properties.classes()[0].len(), 4);
```

Comment:
I suggest actually validating the contents of these equivalence classes somehow (rather than just their sizes)

Reply:
Sure, will do.